### ISSUE TITLE
TypeError: Invalid Version: undefined when initializing angular 12 project

### ISSUE THREAD

**Issue by tl24** [storybookjs/storybook#15260]
**Describe the bug**
I get the following error "TypeError: Invalid Version: undefined" when trying to run "sb init -t angular -N" on an angular 12 project.  I also get the same error if running "npx sb init".  This is running sb version 6.3.0-rc.7.

**To Reproduce**
Install angular as a devDependencies rather than dependencies (this is for a library project)

**System**
Environment Info:

  System:
    OS: Linux 5.4 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
  Binaries:
    Node: 14.17.0 - /home/linuxbrew/.linuxbrew/opt/node@14/bin/node
    Yarn: 1.22.10 - /home/linuxbrew/.linuxbrew/bin/yarn
    npm: 6.14.13 - /home/linuxbrew/.linuxbrew/opt/node@14/bin/npm

**Additional context**

**Comment by shilman** [comment]
what happens if you run `npx sb@next init`?

**Comment by Marklb** [comment]
I may be able to try this later, but if someone else looks at it I assume it is this line: https://github.com/storybookjs/storybook/blob/next/lib/cli/src/generators/ANGULAR/index.ts#L37

**Comment by thiagosbrito** [comment]
@shilman For me, even using `npx sb@next init` I get the same issue.

**Comment by fmanadi** [comment]
I had the same error when trying to initialize storybook inside an angular 12.1.0 library project. `I ran npx sb@next init` and got the same error again.
It appears that storybook initialization script was expecting the `@angular/core`  dependency to be listed in my `package.json` in the "**dependencies**" section, but mine was listed in the "**peerDependencies**" instead.
So I moved the `@angular/core` from **peerDependencies** to **dependencies** and it worked for me.

**Comment by tl24** [comment]
> I may be able to try this later, but if someone else looks at it I assume it is this line: https://github.com/storybookjs/storybook/blob/next/lib/cli/src/generators/ANGULAR/index.ts#L37

Yes, I think that is the issue.  I was able to temporarily move @angular/core into dependencies in the package.json to work around it.

**Comment by thiagosbrito** [comment]
@tl24 which `package.json`? Application or storybook one?

**Comment by fmanadi** [comment]
@thiagosbrito The `package.json` of the angular app, not the storybook one.

**Comment by thiagosbrito** [comment]
Thanks @tl24 & @fmanadi, this solution fixed my issue. After installation I removed angular/core from package.json

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid.

Please open a new issue referencing this one if:
- this is still a problem in SB 7.x
- you can provide a consistent reproduction in 7.x (at https://storybook.new/ or per [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce)).
